dk.sdu.se4.weapon/src/main/java/dk/sdu/se4/weapon/WeaponProcessor.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package dk.sdu.se4.weapon;
+
+import dk.sdu.se4.common.entity.Entity;
+import dk.sdu.se4.common.entity.part.MovingPart;
+import dk.sdu.se4.common.entity.part.WeaponPart;
+import dk.sdu.se4.common.service.ProcessorService;
+import dk.sdu.se4.commongameinput.GameInput;
+import dk.sdu.se4.commongameinput.GameInputKeys;
+
+/**
+ *
+ * @author steff
+ */
+public class WeaponProcessor extends WeaponCore implements ProcessorService {
+    
+    private GameInput gameInput = null;
+    
+    public void addGameInput(GameInput gameInput) {
+        this.gameInput = gameInput;
+    }
+    
+    public void removeGameInput(GameInput gameInput) {
+        this.gameInput = null;
+    }
+    
+    @Override
+    public void process() {
+        if (this.mapService != null) {
+            for (Entity e : this.mapService.getEntities(Weapon.class)) {
+                MovingPart mp = e.getPart(MovingPart.class);
+                WeaponPart wp = e.getPart(WeaponPart.class);
+                mp.process(e);
+                if(this.gameInput.isPressed(GameInputKeys.SPACE)) {
+                    //TODO use wp firerate for shooting interval
+                    if (wp.getType()) {
+                        if(wp.getTotalBullets() == 0) {
```

Comment:
I now see where you used "totalBullets", but shouldn't this be implemented in the part instead of here?

Reply:
https://github.com/Nicklason/4-semester-project/pull/54#discussion_r611252922